tico/serialize/operators/op_avg_pool2d.py

```diff
@@ -26,16 +27,79 @@
 from tico.serialize.operators.node_visitor import NodeVisitor, register_node_visitor
 from tico.serialize.operators.utils import create_builtin_operator, get_op_index
 from tico.utils.define import define_pad_node
+from tico.utils.errors import NotYetSupportedError
 from tico.utils.validate_args_kwargs import AvgPool2dArgs
 
 
 @register_node_visitor
 class AvgPool2DVisitor(NodeVisitor):
+    """
+    This class defines how to serialize AvgPool2D operation into Circle IR.
+
+    Torch                           | Circle
+
+    count_include_pad: True/False   | (count_include_pad): Always False
+    (padding): Always "valid"       | padding: "valid"/"same"
+
+    * Circle's avgpool2d has no option for count_include_pad, so we always set it as False.
+    * Torch's avgpool2d always uses "valid" padding mode.
+    """
+
     target: List[torch._ops.OpOverload] = [torch.ops.circle_custom.avgpool2d]
 
     def __init__(self, op_codes: Dict[OpCode, int], graph: CircleSubgraph):
         super().__init__(op_codes, graph)
 
+    def has_padding(self, args: AvgPool2dArgs) -> bool:
+        padding = args.padding
+        if padding[0] == 0 and padding[1] == 0:
+            return False
+        else:
+            return True
+
+    def has_same_padding(self, args: AvgPool2dArgs) -> bool:
+        input_shape = list(extract_shape(args.input))
+        kernel_size = args.kernel_size
+        stride = args.stride
+        padding = args.padding
+
```

Comment:
The below equation only works when ceil_mode == False. Better to add a guard here

```suggestion
        # TODO Update this function when supporting ceil_mode = True
        assert args.ceil_mode is False
```